### DISCUSSION TITLE
Realtime not working on local

### DISCUSSION THREAD

**Discussion by MahmoodKhalil57** [pocketbase/pocketbase#1515] [Q&A]
Realtime data only works on pockethost.io but it is giving the following error when I try to host same exact collections but on local.
 Do note that normal CRUD calls work fine on both local pocketbase and pockethost.io fine,

![image](https://user-images.githubusercontent.com/46604753/211121237-a33ea8ae-ddad-4dcd-8983-2e6a40c818ce.png)

**Comment by ganigeorgiev** [comment]
It seems like a CORS issue. PocketBase by default allows all origins (aka. `*`) so my guess is that this is caused by your node server (aka. the one that serves `localhost:3000`).

**Reply by ganigeorgiev** [reply]
If that's not the issue, try wrapping your `subscribe()` calls with `try/catch` to inspect the error:

```js
try {
    await pb.collection('example').subscribe('*', () => { ... });
} catch(err) {
    console.log(err.originalError);
}
```

**Reply by MahmoodKhalil57** [reply]
Oh wow you reply fast, Im using sveltekit and im not setting any CORS rules on my node server or pocketbase server

I will do the try catch and paste the error and Ill try to create a minimum environment incase you want to test it yourself.

**Reply by ganigeorgiev** [reply]
@MahmoodKhalil57 _vite has released a new version since then and it may not be the case anymore, but your issue could be related to https://github.com/sveltejs/kit/issues/6455, aka. check your vite.config and see if you are using different cors setting than the defaults._

**Reply by MahmoodKhalil57** [reply]
I checked the config, and there is nothing there about CORS

This is the error message Im getting when I do your console log
![image](https://user-images.githubusercontent.com/46604753/211147508-55662c1c-4864-47ed-baed-4eb50acec3f6.png)


Also this is a project you can use to test it out
https://github.com/MahmoodKhalil57/svelteRealtimeTest
to run the project:
    pnpm install
    pnpm run dev
    ./pocketbase/pocketbase serve

admin panel cred:
    test@user.com
    password00

**Reply by ganigeorgiev** [reply]
I'm not able to reproduce the issue with the provided repo. The realtime events are working as expected for me (Node 18.10.0 and PocketBase v0.10.4):

![inspect_tool_screenshot](https://i.imgur.com/MAPHlkT.png)

**Reply by MahmoodKhalil57** [reply]
That is very weird, I tried with a different system and I tried 1.18.10. and Im still getting the same error. 
Did you change anything about pocketbase? Did you make any changes? Did you use windows? Did you use pnpm?

I ran the following on the different computer and got the same exact issue (both my tests are on windows 11)

node 1.18.10
npm 8.19.3
pnpm 7.17.1
pocketbase 0.10.4

git clone https://github.com/MahmoodKhalil57/svelteRealtimeTest
pnpm install
pnpm run dev
./pocketbase/pocketbase serve

**Reply by ganigeorgiev** [reply]
> Did you change anything about pocketbase? 

My test was with a local PocketBase v0.10.4 executable but with `--dir` `--migrationsDir` pointing to the ones attached in the repo.

> Did you make any changes?

No. The only thing that out of the ordinary was an issue while running `npm install` related to conflicting "uno" dependency but was resolve with `--legacy-peer-deps`.

> Did you use windows? 

No, I'm using Fedora/linux.

> Did you use pnpm?

No, npm, but I don't see why this should cause an issue (the request is executed in the browser and as long as the node server is serving the frontend, it should be fine).

---

I'm not sure why you are getting the error, but I don't think it is related to PocketBase.
You can try accessing your frontend with the same host, aka. `127.0.0.1` instead of `localhost` and see if that makes any difference.
It is possible that there is something wrong with your windows configuration, the hosts file or some firewall is intercepting the request, I'm not really sure.

**Reply by MahmoodKhalil57** [reply]
I understand that I may not be providing enough information, Im not very techy so im not sure what info I should provide, would you be able to join a public discord channel like svelte's and i can share my screen and follow your instructions? I can later update this thread with a log of what I was instructed to do.

**Reply by ganigeorgiev** [reply]
No, sorry. You can try to:

- Access your frontend with `127.0.0.1:3000` instead of localhost to rule out hosts file misconfiguration.

- Open the browser's "Inspect tool > Network" tab and check the failed `realtime` entry request and response data.

- Test in incognito mode or with different browser(s) to rule out extension interference.

**Reply by MahmoodKhalil57** [reply]
That's alright,

That's alright

> No, sorry. You can try to:
> 
>     * Access your frontend with `127.0.0.1:3000` instead of localhost to rule out hosts file misconfiguration.

       I tried this, still doesnt work
![image](https://user-images.githubusercontent.com/46604753/211152510-29c3c2ad-4217-4c68-ad7f-bbf2325410b0.png)

> 
>     * Open the browser's "Inspect tool > Network" tab and check the failed `realtime` entry request and response data.
the request and response are both empty
![image](https://user-images.githubusercontent.com/46604753/211152544-9b042d39-b262-4142-9f95-fd75be144adb.png)
![image](https://user-images.githubusercontent.com/46604753/211152578-e0a24d6e-bfa8-41fa-8c42-c2e9adb035a6.png)
![image](https://user-images.githubusercontent.com/46604753/211152588-2650f49e-aa58-49b0-aae4-69cbb00c5288.png)

> 
>     * Test in incognito mode or with different browser(s) to rule out extension interference.

Doesnt work, I even tried different browsers.

I guess Ill have to use pockethost.io for testing for now since that works, and if I test this later and get it to work Ill update this thread.

**Reply by MahmoodKhalil57** [reply]
Final update for a while: I downloaded WSL Ubuntu 20.04 LTS and pocketbase for linux, and ran both the front end and pocketbase, and tried to use npm --legacy-peer-deps instead of pnpm, the issue is still unresolved, I am still getting the same errors

![image](https://user-images.githubusercontent.com/46604753/211170893-f5a4b607-0258-4447-98ba-b1a920739e4d.png)

(note: Im working on small projects so this is just a minor inconvenience to me, I can just use pockethost.io for testing purposes, and all in all, pocketbase is a pleasure to deal with)

**Reply by ganigeorgiev** [reply]
I'm not able to reproduce it and I don't think the issue is related to PocketBase.

By default all origins are allowed, so the above error is caused by something in your local environment (firewall, antivirus software, browser extension, browser flags, etc.).

**Reply by ganigeorgiev** [reply]
The only way I'm able to reproduce `/api/realtime` error (which eventually could result to the above CORS error because the request wouldn't complete) is if you have multiple tabs opened with the realtime page. This is because when using SSE locally with HTTP1 it is limited by the browsers to ~6 connections (check the note in https://developer.mozilla.org/en-US/docs/Web/API/Server-sent_events/Using_server-sent_events).

**Reply by MahmoodKhalil57** [reply]
I closed all firefox tabs and opened a single incognito chrome tab and I did in fact stop getting that cors error.


I am now getting a different one. I will test a bit more in the morning on different computer and using WSL and comment the results.
![image](https://user-images.githubusercontent.com/46604753/211172501-c4f2e515-7d00-4cc3-8f27-504ba2536f4b.png)
![image](https://user-images.githubusercontent.com/46604753/211172566-5856f237-8ba0-4ed4-a7da-23cb14c9f031.png)

**Comment by MahmoodKhalil57** [comment]
After a long thread, here is how I fixed my CORS issue

@ganigeorgiev -
     "The only way I'm able to reproduce /api/realtime error (which eventually could result to the above CORS error because the request wouldn't complete) is if you have multiple tabs opened with the realtime page. This is because when using SSE locally with HTTP1 it is limited by the browsers to ~6 connections (check the note in https://developer.mozilla.org/en-US/docs/Web/API/Server-sent_events/Using_server-sent_events)."

(edit)(edit)

Listening to his advice fixed the issue when testing on chrome, but testing on firefox still occasionally got the same CORS error, but on both chome and firefox, realtime is not working in my setup regardless of CORS.
![image](https://user-images.githubusercontent.com/46604753/211173373-7cf4ef57-b287-4a6f-9868-93217ad7d9b7.png)

**Comment by Irvingouj** [comment]
I get similar error on the real time api, but only on firefox
```
Cross-Origin Request Blocked: The Same Origin Policy disallows reading the remote resource at http://localhost:8090/api/realtime. (Reason: CORS request did not succeed). Status code: (null).
```

I am pretty sure I have one firefox and one tab running, the method is called only once. how can I fix?

**Reply by ganigeorgiev** [reply]
@Irvingouj This is a very old discussion and I'm not sure how your issue is related to the above.

In any case, PocketBase by default allows all origins so if you have CORS issue it could be caused by a reverse proxy (nginx, apache, etc.), browser extension, misconfigured 3rd-party app (Netxt.js, SvelteKit, etc.).

If you are still not able to resolve your issue, please open a new Q&A discussion with more details about your project and ideally some minimal reproducible steps.
